dds/idl/rapidjson_generator.cpp

```diff
@@ -264,17 +267,22 @@ namespace {
           dds_generator::scoped_helper(type->name(), "_"),
           temp_type = scoped(type->name()),
           temp_name = "temp_" + underscores;
+        const bool octet_type = (pt == AST_PredefinedType::PT_octet);
         strm <<
           ip << "if (val.IsString()) {\n" <<
           ip << "  std::string ss(val.GetString());\n" <<
           ip << "  std::istringstream iss(ss);\n" <<
-          ip << "  " << (pt == AST_PredefinedType::PT_octet ? "uint16_t" : temp_type.c_str()) << " " << temp_name << ";\n" <<
+          ip << "  " << (octet_type ? "uint16_t" : temp_type.c_str()) << " " << temp_name << ";\n" <<
           ip << "  if (ss.find(\"0x\") != std::string::npos) {\n" <<
           ip << "    iss >> std::hex >> " << temp_name << ";\n" <<
           ip << "  } else {\n" <<
           ip << "    iss >> " << temp_name << ";\n" <<
           ip << "  }\n" <<
-          ip << "  " << propName << assign_prefix << temp_name << assign_suffix << ";\n" <<
+          ip << "  " << propName << assign_prefix;
+        if (octet_type) {
```

Comment:
I'm guessing it was complaining about truncated values here going from uint16_t to CORBA::Octet. We're only using uint16_t to avoid issues with istringstream and unsigned char (wants to read a character, not a number), but theoretically someone could pass in a value over 255, in which case we just want to cap it at 255. I don't know that the code / tests enforce that though. I'm fine with this change, but in the future we may want an assert or forced cap here or something.

Reply:
True.  Casting isn't the best...